src/network/taraxa_capability.cpp

```diff
@@ -378,23 +393,60 @@ bool TaraxaCapability::interpretCapabilityPacketImpl(NodeID const &_nodeID, unsi
+
           uint64_t peer_pbft_round = _r[0].toPositiveInt64();
+          size_t peer_pbft_previous_round_next_votes_size = _r[1].toInt<unsigned>();
           uint64_t pbft_round = pbft_mgr_->getPbftRound();
-          if (pbft_round > peer_pbft_round) {
-            LOG(log_dg_next_votes_sync_) << "Current PBFT round is " << pbft_round << ", and peer PBFT round is "
-                                         << peer_pbft_round << ". Will send bundle of next votes";
+          size_t pbft_previous_round_next_votes_size = next_votes_mgr_->getNextVotesSize();
+
+          if (pbft_round > peer_pbft_round ||
+              (pbft_round == peer_pbft_round &&
+               pbft_previous_round_next_votes_size > peer_pbft_previous_round_next_votes_size)) {
+            LOG(log_dg_next_votes_sync_) << "Current PBFT round is " << pbft_round << " previous round next votes size "
+                                         << pbft_previous_round_next_votes_size << ", and peer PBFT round is "
+                                         << peer_pbft_round << " previous round next votes size "
+                                         << peer_pbft_previous_round_next_votes_size
+                                         << ". Will send out bundle of next votes";
             sendPbftNextVotes(_nodeID);
           }
+
           break;
         }
         case PbftNextVotesPacket: {
           auto next_votes_count = _r.itemCount();
-          LOG(log_dg_next_votes_sync_) << "In PbftNextVotesPacket received " << next_votes_count << " next votes";
+          if (next_votes_count == 0) {
+            LOG(log_er_next_votes_sync_) << "Synced 0 next votes from peer " << _nodeID
+                                         << ". The peer may be a malicous player, will be disconnected";
+            host_.capabilityHost()->disconnect(_nodeID, p2p::UserReason);
+
+            break;
+          }
+          LOG(log_nf_next_votes_sync_) << "Received " << next_votes_count << " next votes from peer " << _nodeID;
+
+          std::vector<Vote> next_votes;
           for (auto i = 0; i < next_votes_count; i++) {
             Vote next_vote(_r[i].data().toBytes());
-            LOG(log_dg_next_votes_sync_) << "Received PBFT next vote " << next_vote.getHash();
-            vote_mgr_->addVote(next_vote);
+            LOG(log_nf_next_votes_sync_) << "Received PBFT next vote " << next_vote.getHash();
+
+            next_votes.emplace_back(next_vote);
+          }
+
+          auto pbft_current_round = pbft_mgr_->getPbftRound();
+          auto pbft_previous_round_next_votes_size = next_votes_mgr_->getNextVotesSize();
+          auto peer_pbft_round = next_votes[0].getRound() + 1;
+
+          if (pbft_current_round < peer_pbft_round) {
+            // Add into votes unverified queue
+            vote_mgr_->addVotes(next_votes);
+          } else if (pbft_current_round == peer_pbft_round && pbft_previous_round_next_votes_size < next_votes_count) {
```

Comment:
What if our node has 3 votes and other node has 2 votes, but those two votes are different than our 3?

Reply:
if our round < peer round, put into unverified queue, and will check the vote hash is exist
if our round == peer round && our previous next votes size < peer, in this case, at least they will have (t+1) same next votes. And our node adds the missing next votes